jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java

```diff
@@ -283,6 +285,19 @@ public void setInheritChannel(boolean inheritChannel)
         _inheritChannel = inheritChannel;
     }
 
+    private void setAcceptChannel(ServerSocketChannel acceptChannel) throws IOException
+    {
+        _acceptChannel = acceptChannel;
+        _localPort = _acceptChannel.socket().getLocalPort();
+        SocketAddress localAddress = _acceptChannel.socket().getLocalSocketAddress();
+        if (localAddress instanceof InetSocketAddress)
+            _localName = ((InetSocketAddress)localAddress).getHostString();
+        else
+            _localName = localAddress.toString();
+        if (_localPort <= 0)
+            throw new IOException("Server channel not bound");
```

Comment:
It is kind of strange that the `localAddress` here is used to initialize the `_localName`, but not the `_localAddress` or `_serverAuthority`.

I know you said there is a reason??? was it the Proxy protocol?   So either we need to make another attempt to initialize these here or we have to document somewhere in the code why not.   If they are not going to be initialized, then we need better javadoc on getLocalAddress to say what null return means

Reply:
I reverted this.  It's left over from when we attempted to default the localAddress during doStart().
That (the defaulting of localAddress at doStart) couldn't be done for the proxy cases.